core/deployment/src/main/java/io/quarkus/deployment/steps/RuntimeConfigSetupBuildStep.java

```diff
@@ -37,11 +40,17 @@ void setupRuntimeConfig(
                 .interfaces(StartupTask.class).build()) {
 
             try (MethodCreator method = clazz.getMethodCreator("deploy", void.class, StartupContext.class)) {
-                method.invokeVirtualMethod(ofMethod(StartupContext.class, "setCurrentBuildStepName", void.class, String.class),
+                TryBlock tryBlock = method.tryBlock();
+                tryBlock.invokeVirtualMethod(
+                        ofMethod(StartupContext.class, "setCurrentBuildStepName", void.class, String.class),
                         method.getMethodParam(0), method.load("RuntimeConfigSetupBuildStep.setupRuntimeConfig"));
 
-                method.invokeStaticMethod(C_CREATE_RUN_TIME_CONFIG);
-                method.returnValue(null);
+                tryBlock.invokeStaticMethod(C_CREATE_RUN_TIME_CONFIG);
+                tryBlock.returnValue(null);
+
+                CatchBlockCreator cb = tryBlock.addCatch(Throwable.class);
```

Comment:
Not sure we would want to catch and wrap an `OutOfMemoryError` for example... Unless this is a pattern used elsewhere, I'd recommend catching only `RuntimeException` here.

Reply:
Changed